core/src/main/java/org/apache/struts2/dispatcher/Dispatcher.java

```diff
@@ -561,8 +561,16 @@ public void serviceAction(HttpServletRequest request, HttpServletResponse respon
             String name = mapping.getName();
             String method = mapping.getMethod();
 
-            ActionProxy proxy = getContainer().getInstance(ActionProxyFactory.class).createActionProxy(
-                    namespace, name, method, extraContext, true, false);
+            ActionProxy proxy;
+
+            //check if we are probably in an async resuming
+            ActionInvocation inv = ActionContext.getContext().getActionInvocation();
+            if (inv == null || inv.isExecuted()) {
+                proxy = getContainer().getInstance(ActionProxyFactory.class).createActionProxy(namespace, name, method,
```

Comment:
Name `inv` is a little short for my taste

Reply:
Thank you so much for your time and attention. I'll fix both on my pleasure :)